src/Symfony/Component/Intl/Currencies.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Intl;
+
+use Symfony\Component\Intl\Exception\MissingResourceException;
+
+/**
+ * Gives access to currency-related ICU data.
+ *
+ * @author Bernhard Schussek <bschussek@gmail.com>
+ * @author Roland Franssen <franssen.roland@gmail.com>
+ */
+final class Currencies extends ResourceBundle
+{
+    private const INDEX_SYMBOL = 0;
+    private const INDEX_NAME = 1;
+    private const INDEX_FRACTION_DIGITS = 0;
+    private const INDEX_ROUNDING_INCREMENT = 1;
+
+    /**
+     * @return string[]
+     */
+    public static function getCurrencyCodes(): array
```

Comment:
Why is this method called `getCurrencyCodes()` instead of `getCodes()` ? It looks a bit redundant.

Same question in other classes such as Languages. Thanks.

Reply:
This was discussed above, you can expand the conversations to follow them. "Currency Code", "Language Code" etc. is domain language of CLDR (e.g. https://www.xe.com/iso4217.php). It doesn't make sense to shorten them to "code", which could mean virtually anything.

Second, for many of these things we have different kinds of code representations: 3-letter codes, 2-letter codes, numeric codes. For currencies, it is common to address them by the 3-letter codes. Hence the 3-letter code is refered to by "currency code" here. If you need the numeric code for some reason, you can convert the currency code to it using `Currencies::getNumericCode('EUR')`.

The system is followed by the other resource bundles.

Reply:
Sorry, I didn't follow the previous discussions.

However, now that we have classes for each thing (Currencies, Languages, etc.) adding that into method names looked unnecessary to me. Also, it's a bit inconsistent:

```php
// we do this:
Currencies::getCurrencyCodes()
// instead of this:
Currencies::getCodes()

// but we do this:
Currencies::getNames()
// instead of this:
Currencies::getCurrencyNames()
```

But consistency with the upstream standard/library/tech is mor important 🙏  So let's keep it.

Reply:
at least "name" is not ambiguous IMHO, it always means a "currency name".

Reply:
@javiereguiluz That's if you look at it from a technical perspective :) we are programmers, and we want to remove redundancies, of course.

From a domain perspective, however, this makes sense. We have a currency, and for the currency we have its
* name
* symbol
* currency code (not "code")
* numeric code
* rounding increment
* fraction digits
* exchange rate
* ...

Reply:
"code" also always means "currency code". But we might have to deal with different type of codes, within the currency domain (whereas we never deal with different types of names)